umbrella/router.go

```diff
@@ -57,6 +58,20 @@ type funcRoutingInfo struct {
 	Manual      string
 }
 
+// Function to compare a lowercase version of the requested slack command to a lowercase registry command
+// Probably doesn't work because you would have to check against every command in the registry and transform them
+// Also, not sure where to call it. Presumably within the first few error checks of getFuncRoutingInfo()
+func compareCmdToReg(reqCmd string, regCmd string) {
```

Comment:
Monsieur, where does this function get called in the flow of things?

Reply:
Other than (presumably) early in getFuncRoutingInfo(), I really have no idea. I'm having trouble deciphering where Slack commands and Registry commands are coming in from and getting compared. 

I would *think* it would come in somewhere around what is now line 87 or 93. Maybe replacing or working with:
```
cmdInfo, cmdInfoExists := (*cmdReg)[cmd]
	if !cmdInfoExists {
		return nil, fmt.Errorf("Command is not registered. Command Name: '%s'", cmd)
	}
```
But my Go is too poor to figure out where 1) the passed slack argument comes in, and 2) where/how you check it against function names in the registry. Not being able to figure that out, I don't know where to get the arguments to pass to it, a la:
```
if !compareCmdToReg(slackCmd, regCmd) {
    return nil, fmt.Errorf("etc")
}
```

Reply:
Also, I realize you need to compare the slack command against the registry entries, so in my a la example above, you would have presumably already solved the problem and made the comparison before that happened, hehe.

Reply:
Assuming `cmd` from line 87 is the passed command, (`/struggle lowPriority` <--- this one).

Not sure what's happening after that with the cmdReg map, or how to do a `strings.ToLower(cmdReg[strings.ToLower(cmd)])` kind of check.

Reply:
The entry-point for loading the registry is here: https://github.com/junior-dev-struggle-bus/struggle-slack-app/blob/d6abc212b46db36f48d4d7738ce2890a274beafb/umbrella/router.go#L267

This parses out 'struggle' from '/struggle': https://github.com/junior-dev-struggle-bus/struggle-slack-app/blob/d6abc212b46db36f48d4d7738ce2890a274beafb/umbrella/router.go#L72

This pulls out 'lowPriority' from the Slack arguments: https://github.com/junior-dev-struggle-bus/struggle-slack-app/blob/d6abc212b46db36f48d4d7738ce2890a274beafb/umbrella/router.go#L92

Your problem here is that if someone registers their function like I did for you as 'lowPriority', it gets unmarshaled into the [routing object](https://github.com/junior-dev-struggle-bus/struggle-slack-app/blob/d6abc212b46db36f48d4d7738ce2890a274beafb/umbrella/router.go#L51-L58) as, 'lowPriority'. So, you'd have to walk the map after the registry is loaded and standardize how things are loaded into there. This is where the registry gets unmarshaled: https://github.com/junior-dev-struggle-bus/struggle-slack-app/blob/d6abc212b46db36f48d4d7738ce2890a274beafb/umbrella/router.go#L125

Reply:
Also, case-insensitive mapping idea: https://stackoverflow.com/questions/11124705/how-to-create-a-case-insensitive-map-in-go

Reply:
Wrote a function to use strings.EqualFold to compare against registered functions. Should return the function name as it appears in the registry to `confirmedFuncName` and then check against that as you did before with `funcName`.

Not 100% confident on my syntax/pointers/error reporting, but tried to model what you did elsewhere in the code.